docker/README.md

```diff
@@ -0,0 +1,480 @@
+# Fluster Docker Setup
+
+Docker configuration for Fluster with comprehensive codec support including **FFmpeg** and **GStreamer** with hardware acceleration support (**VAAPI**, **VDPAU**, **NVDEC**, **QuickSync**).
+
+## Features
+
+- **Customizable Ubuntu version**: 20.04, 22.04, or 24.04
+- **Three FFmpeg installation options**:
+  1. **System packages** (default) - Fast build, but older versions
+  2. **Static binaries** (7.x, 8.x) - Newer FFmpeg, but limited backend support
+  3. **Build from source** (recommended) - Latest FFmpeg with all backends enabled
+- **Hardware Acceleration Support**:
+  - **VAAPI** (Intel, AMD, NVIDIA on Ubuntu 24.04)
+  - **VDPAU** (Intel via VA-GL bridge, AMD, NVIDIA)
+  - **NVDEC/NVENC** (NVIDIA)
+  - **Intel QuickSync** (libmfx, libmfx-gen)
+
+## FFmpeg Installation Options
+
+This Docker setup provides three ways to install FFmpeg, each with different trade-offs:
+
+### Option 1: System Packages (Default)
+
+**Use case**: Quick testing, CI/CD pipelines
+**Pros**: Fast build time (~5 minutes)
+**Cons**: Older FFmpeg versions, limited backend support
+
+| Ubuntu | FFmpeg | Backend Support                           |
+|--------|--------|-------------------------------------------|
+| 20.04  | 4.2.x  | Limited (basic VAAPI, VDPAU)              |
+| 22.04  | 4.4.x  | Good (VAAPI, VDPAU, some NVDEC/QuickSync) |
+| 24.04  | 6.1.x  | Best (VAAPI, VDPAU, NVDEC, QuickSync)     |
+
+```bash
+# Build with system packages (default)
+./run-docker.sh --rebuild list -c
+```
+
+### Option 2: Static Binaries
+
+**Use case**: Testing specific FFmpeg versions quickly
+**Pros**: Easy to use, newer FFmpeg versions available
+**Cons**: Limited hardware backend support (no QuickSync/libmfx)
+
+```bash
+# Build with FFmpeg 7.0 static binary
+docker build --build-arg FFMPEG_STATIC_VERSION=7.0 -t fluster:ffmpeg7 -f docker/Dockerfile .
+
+# Build with FFmpeg 8.0 static binary
+docker build --build-arg FFMPEG_STATIC_VERSION=8.0 -t fluster:ffmpeg8 -f docker/Dockerfile .
+```
+
+**Note**: Static binaries from johnvansickle.com lack libmfx (QuickSync) support.
+
+### Option 3: Build from Source (Recommended)
+
+**Use case**: Full hardware acceleration testing, production use
+**Pros**: Latest FFmpeg 8.0 with ALL backends enabled (VAAPI, VDPAU, QuickSync)
+**Cons**: Longer build time (~20-30 minutes)
+
+```bash
+# Build FFmpeg 8.0 from source with all backends
+./run-docker.sh --build-ffmpeg --rebuild list -c
+
+# Or manually specify version
+docker build \
+  --build-arg BUILD_FFMPEG_FROM_SOURCE=1 \
+  --build-arg FFMPEG_VERSION=8.0 \
+  -t fluster:ffmpeg-full \
+  -f docker/Dockerfile .
+```
+
+**Included backends when building from source:**
```

Comment:
A couple of doubts about the list below.
With this installation mode do we lose support for Nvidia when using VAAPI and for Intel when using VDPAU?